crates/shared/src/bad_token/token_owner_finder/seasolver.rs

```diff
@@ -0,0 +1,110 @@
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
+    time::Duration,
+};
+
+use anyhow::Result;
+use ethcontract::H160;
+use reqwest::{Client, Url};
+
+use super::TokenOwnerProposing;
+
+type Token = H160;
+type Owner = H160;
+
+#[derive(Clone, Debug)]
+pub struct SeaSolverConfiguration {
+    pub url: Url,
+    pub client: Client,
+    pub update_interval: Duration,
+}
+
+impl SeaSolverConfiguration {
+    /// Return type is `Token, Option<Owner>` because there are
+    /// entries containing `Null` instead of owner address.
+    async fn query(&self) -> Result<HashMap<Token, Option<Owner>>> {
```

Comment:
Would the API scale better if they had something like:
```
https://seasolver.dev/token_holders?token=0x....
```

Reply:
Might be, but I guess the caching would be more complex to implement than... Not sure if worth it